src/js/app.js

```diff
@@ -374,7 +375,11 @@ define(function(require) {
     },
 
     _getSubtotal: function() {
-      return this._getPrice() * this._getQuantity();
+      var linetotal = this._getPrice() * this._getQuantity();
+      var discount = this._getDiscount();
+      var subtotal = Math.max(linetotal - discount, 0);
```

Comment:
I think `_getSubtotal` should not include subtracting the discount. If you want to make a convenience method, we should make a `_getSubtotalMinusDiscount`. The reason being that the subtotal is not inherently after discount. `_getTaxes` would have to be updated to use the new method.